core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3931,6 +3939,34 @@ public static List distinct(List list) {
     return min;
   }
 
```

Comment:
comments:
it can not use 

```
list.add
return list;
```
because it will throw exception
java.lang.UnsupportedOperationException
	at java.util.AbstractList.add(AbstractList.java:148)
	at java.util.AbstractList.add(AbstractList.java:108)
	at org.apache.calcite.runtime.SqlFunctions.arrayAppend(SqlFunctions.java:3899)